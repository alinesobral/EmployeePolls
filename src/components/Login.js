import { useState } from "react";
import { handleLogin } from "../actions/authedUser";
import { connect } from "react-redux";

const Login = (props) => {
  const [usernameTxt, setUsernameTxt] = useState("");
  const [passwordTxt, setPasswordTxt] = useState("");
  const [loginError, setLoginError] = useState(false);

  const handleChangeUser = (e) => {
    const text = e.target.value;
    setUsernameTxt(text);
  };
  const handleChangePwd = (e) => {
    const text = e.target.value;
    setPasswordTxt(text);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    props.dispatch(handleLogin(usernameTxt, passwordTxt)).then((response) => {
      if (!response) {
        setLoginError(true);
      }
    });
    setUsernameTxt("");
    setPasswordTxt("");
  };
  return (
    <div className="box" data-testid="login">
      <h1>Employee Polls</h1>
      {loginError && <p data-testid="loginError">Invalid username/password</p>}
      <form onSubmit={handleSubmit}>
        <div>
          <p>Username: </p>
          <input
            placeholder="Username"
            value={usernameTxt}
            onChange={handleChangeUser}
            className="textbox"
            maxLength={32}
            data-testid="username"
          />
        </div>
        <div>
          <p>Password: </p>
          <input
            placeholder="Password"
            type="password"
            value={passwordTxt}
            onChange={handleChangePwd}
            className="password"
            maxLength={32}
            data-testid="password"
          />
        </div>
        <div>
          <button
            type="submit"
            className="btn"
            disabled={usernameTxt === "" && passwordTxt === ""}
            data-testid="loginUserBtn"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  );
};

export default connect()(Login);
