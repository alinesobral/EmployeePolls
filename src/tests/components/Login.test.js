import { Provider } from "react-redux";
import { legacy_createStore } from "redux";
import { MemoryRouter } from "react-router-dom";
import { render, screen, fireEvent } from "@testing-library/react";

import reducer from "../../reducers";
import middleware from "../../middleware";
import Login from "../../components/Login";

const store = legacy_createStore(reducer, middleware);
console.log(store.getState());

describe("Login", () => {
  it("will show the login page", () => {
    var component = render(
      <Provider store={store}>
        <MemoryRouter>
          <Login />
        </MemoryRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it("will log the user in", () => {
    var component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );

    var username = screen.getByTestId("username");
    fireEvent.change(username, { target: { value: "sarahedo" } });
    var password = screen.getByTestId("password");
    fireEvent.change(password, { target: { value: "password123" } });
    var loginUser = screen.getByTestId("loginUserBtn");
    fireEvent.click(loginUser);

    expect(component.getByTestId("dashboard")).toBeInTheDocument();
  });

  it("will show an alert for an unsuccessful authentication", () => {
    var component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Login />
        </Provider>
      </MemoryRouter>
    );

    var username = screen.getByTestId("username");
    fireEvent.change(username, { target: { value: "sarahedo" } });
    var password = screen.getByTestId("password");
    fireEvent.change(password, { target: { value: "wrongPassword" } });
    var loginUser = screen.getByTestId("loginUserBtn");
    fireEvent.click(loginUser);

    expect(component).toMatchSnapshot();
  });
});
