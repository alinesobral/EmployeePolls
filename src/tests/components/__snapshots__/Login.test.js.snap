// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login will show an alert for an unsuccessful authentication 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="box"
      >
        <h1>
          Employee Polls
        </h1>
        <form>
          <div>
            <p>
              Username: 
            </p>
            <input
              class="textbox"
              data-testid="username"
              maxlength="32"
              placeholder="Username"
              value=""
            />
          </div>
          <div>
            <p>
              Password: 
            </p>
            <input
              class="password"
              data-testid="password"
              maxlength="32"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div>
            <button
              class="btn"
              data-testid="loginUserBtn"
              disabled=""
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="box"
    >
      <h1>
        Employee Polls
      </h1>
      <form>
        <div>
          <p>
            Username: 
          </p>
          <input
            class="textbox"
            data-testid="username"
            maxlength="32"
            placeholder="Username"
            value=""
          />
        </div>
        <div>
          <p>
            Password: 
          </p>
          <input
            class="password"
            data-testid="password"
            maxlength="32"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <div>
          <button
            class="btn"
            data-testid="loginUserBtn"
            disabled=""
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Login will show the login page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="box"
      >
        <h1>
          Employee Polls
        </h1>
        <form>
          <div>
            <p>
              Username: 
            </p>
            <input
              class="textbox"
              data-testid="username"
              maxlength="32"
              placeholder="Username"
              value=""
            />
          </div>
          <div>
            <p>
              Password: 
            </p>
            <input
              class="password"
              data-testid="password"
              maxlength="32"
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div>
            <button
              class="btn"
              data-testid="loginUserBtn"
              disabled=""
              type="submit"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="box"
    >
      <h1>
        Employee Polls
      </h1>
      <form>
        <div>
          <p>
            Username: 
          </p>
          <input
            class="textbox"
            data-testid="username"
            maxlength="32"
            placeholder="Username"
            value=""
          />
        </div>
        <div>
          <p>
            Password: 
          </p>
          <input
            class="password"
            data-testid="password"
            maxlength="32"
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <div>
          <button
            class="btn"
            data-testid="loginUserBtn"
            disabled=""
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
